### PR TITLE
AosEdge Unit to Cloud protocol v6

### PR DESCRIPTION
**Pull request by oleh-mykytiuk**
- added JSOM schema describes the 6's version of the protocol
 - autogenerated HTML version of the documentation
 - autogenerated MD version of the documentation

### CODE REVIEW

File: unit-cloud/aos-unit-messages.schema.json [hunk 1]
```diff
@@ -0,0 +1,2657 @@
...
+                    "type": "array"
+                }
+            },
+            "required": [
+                "messageType"
+            ],
+            "title": "AosDesiredStatus",
+            "type": "object"
+        },
+        "AosEnvVar": {
```
Comment (al1img) [inline unit-cloud/aos-unit-messages.schema.json:1047]: Could you group messages by their usage? For example: all env related messages in one place?

File: unit-cloud/aos-unit-messages.schema.json [hunk 2]
```diff
@@ -0,0 +1,2657 @@
...
+            "properties": {
+                "name": {
```
Comment (al1img) [inline unit-cloud/aos-unit-messages.schema.json:1049]: Also should be ID field to match status

File: unit-cloud/aos-unit-messages.schema.json [hunk 3]
```diff
@@ -0,0 +1,2657 @@
...
+                "messageType": {
+                    "const": "installUnitCertificatesConfirmation",
+                    "description": "Message body type.",
+                    "enum": [
+                        "installUnitCertificatesConfirmation"
+                    ],
+                    "title": "Message type",
+                    "type": "string"
+                },
+                "requests": {
```
Comment (al1img) [inline unit-cloud/aos-unit-messages.schema.json:1200]: Requests in confirmation look weird.

File: src/cloud_common/protocols/unit/alert.py [hunk 4]
```diff
@@ -0,0 +1,339 @@
...
+    ]
+
+
+class AosAlertSystemError(AosBaseAlert):
+    """Aos Unit system error alert information."""
+
+    tag: Annotated[
+        Literal['systemAlert'],
+        Field(description='type of the alert'),
+    ]
```
Comment (al1img) [inline src/cloud_common/protocols/unit/alert.py:45]: Do you follow any rule to put empty line between class fields?

File: src/cloud_common/protocols/unit/certificates.py [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+            alias='data',
+            title='Data of the unit secret.',
+            description='Data of the unit secret.',
+        ),
+    ]
+
+
+class AosRenewCertsNotification(BaseModel):
+    """
+    AosUnit protocol: 'renewCertificatesNotification' message.
```
Comment (al1img) [inline src/cloud_common/protocols/unit/certificates.py:66]: Use same tamplate for comments:
"""IssueCertData issue certificate data."""
"""Keeps unit secret used to decode secure device password."""
"""AosUnit protocol: 'issuedUnitCertificates' message."""

File: src/cloud_common/protocols/unit/desired_status.py [hunk 6]
```diff
@@ -0,0 +1,431 @@
...
+    ]
+
+
+class AosDesiredComponentInfo(BaseModel):
+    """Component info sent from the AosEdge Cloud."""
+
+    version: TypeVersionMandatory
+    type: TypeComponentType
+    annotations: TypeComponentAnnotationsOptional
+
```
Comment (al1img) [inline src/cloud_common/protocols/unit/desired_status.py:249]: We agreed to add `id` field` as well. As we may have few components with same type but different versions.

File: src/cloud_common/protocols/unit/alert.py (hunk 4 above)
Comment (oleh-mykytiuk) [reply src/cloud_common/protocols/unit/alert.py:45]: https://peps.python.org/pep-0257/#multi-line-docstrings
https://peps.python.org/pep-0008/#blank-lines

A few words: blank line after docstring
Blank lines to logical grouping

Comment (al1img) [reply src/cloud_common/protocols/unit/alert.py:45]: I see that sometimes you put empty line after multiline field, sometimes no:

```python
   tag: Annotated[
        Literal['coreAlert'],
        Field(description='type of the alert'),
    ]
    node_id: TypeNodeIdMandatory
    core_component: TypeCoreComponentIdMandatory
    message: TypeAlertMessageMandatory
```

```python

    instance: Annotated[
        int,
        Field(
            description='Instance number (starting from 0)',
        ),
    ]

    version: TypeVersionMandatory
    message: TypeAlertMessageMandatory
```

Use one pattern.

File: src/cloud_common/protocols/unit/desired_status.py [hunk 7]
```diff
@@ -0,0 +1,431 @@
...
+#  Copyright (c) 2018-2024 Renesas Inc.
+#  Copyright (c) 2018-2024 EPAM Systems Inc.
+#
+import base64
+from datetime import datetime, time
+from typing import Annotated, Literal
+
+from pydantic import Base64Bytes, BaseModel, Field, field_serializer
+
+from cloud_common.protocols.unit.types import (
```
Comment (al1img) [inline src/cloud_common/protocols/unit/desired_status.py:11]: We discussed that system alerts should be configured by cloud for each node. I don't see this in the protocol.

Another missing thing it is default request resource ration as I remember.

File: src/cloud_common/protocols/unit/monitoring.py [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+    name: Annotated[
+        str,
+        Field(
+            alias='name',
+            description='Name of disk partition',
+        ),
+    ]
+
+    used_size: Annotated[
+        str,
```
Comment (al1img) [inline src/cloud_common/protocols/unit/monitoring.py:30]: int?

File: src/cloud_common/protocols/unit/desired_status.py (hunk 6 above)
Comment (oleh-mykytiuk) [reply src/cloud_common/protocols/unit/desired_status.py:256]: done

**Review by al1img — APPROVED** [review]

**Pull request merged**